kong/core/router.lua

```diff
@@ -446,25 +446,24 @@ function _M.new(apis)
   end)
 
 
-  local grab_headers = #wildcard_hosts > 0 or next(indexes.plain_hosts) ~= nil
+  local grab_host = #wildcard_hosts > 0 or next(indexes.plain_hosts) ~= nil
```

Comment:
This is such a hot code path that the intent here has always been to cache whether retrieving any header is necessary at all. Moving this down to 582 prevents this code to JIT (`next`) and evaluates the same statement over and over again.